lib/thrift/servers/binary_framed/acceptor.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Thrift.Servers.BinaryFramed.Acceptor do
+  @moduledoc """
+  A GenServer that accepts connections on a server and processes the thrift messages.
+  """
+
+  alias Thrift.{
+    Protocol,
+    TApplicationException
+  }
+  use GenServer
+  require Logger
+
+  @spec start_link(pid, module, module) :: GenServer.on_start
+  def start_link(socket, server_module, handler_module) do
+    GenServer.start_link(__MODULE__, [socket, server_module, handler_module])
+  end
+
+  def init([socket, server_module, handler_module]) do
+    Process.send_after(self, :accept, 1)
+    {:ok, {socket, server_module, handler_module}}
+  end
+
+  def handle_info(:accept, {socket, server_module, handler_module}) do
+    {:ok, r_sock} = :gen_tcp.accept(socket)
```

Comment:
I think this will block the entire GenServer until packets are received, which would lock out any kind of operational interaction.